### PR TITLE
test: Fix `WalletTests.GetConflictedOrchardNotes` gtest

### PR DESCRIPTION
**Pull request by str4d**
Also makes various improvements to the testing infrastructure.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK with nonblocking suggestion

**Review by nuttycom — APPROVED** [review]
re-utACK https://github.com/zcash/zcash/pull/5762/commits/1e9bdcd0a2c083501ea14fb3946655b25fc114fb

**Review by therealyingtong — APPROVED** [review]
utACK

File: src/rust/src/transaction_ffi.rs [hunk 1]
```diff
@@ -207,21 +207,32 @@ pub extern "C" fn zcash_transaction_precomputed_init(
...
+                        .unwrap_or_else(|| !all_prev_outputs.is_empty()) =>
                 {
-                    error!("all_prev_outputs is incorrect length");
+                    error!(
+                        "all_prev_outputs is incorrect length {} (should be {})",
+                        all_prev_outputs.len(),
+                        tx.transparent_bundle().map(|t| t.vin.len()).unwrap_or(0),
+                    );
                     return ptr::null_mut();
                 }
```
Comment (daira) [inline src/rust/src/transaction_ffi.rs:237]: This all looks correct, but I think I would have written it something like this:
```rust
Ok(all_prev_outputs) => {
    let expected_len = tx
        .transparent_bundle()
        .map(|t| if t.is_coinbase() { 0 } else { t.vin.len() })
        .unwrap_or(0);
    if all_prev_outputs.len() != expected_len {
        error!(
            "all_prev_outputs is incorrect length {} (should be {}, coinbase={})",
            all_prev_outputs.len(),
            expected_len,
            tx.transparent_bundle().map(|t| t.is_coinbase()).unwrap_or(false),
        );
        return ptr::null_mut();
    }
}
```

File: src/test/test_bitcoin.cpp [hunk 2]
```diff
@@ -74,7 +75,18 @@ BasicTestingSetup::BasicTestingSetup(const std::string& chainName)
...
+    // Uncomment this to log all errors to stdout so we see them in test output.
+    // We don't enable this by default because several tests intentionally cause
+    // verbose error output (while exercising failure cases).
+    // if (pTracingHandle == nullptr) {
+    //     std::string initialFilter = "error";
+    //     pTracingHandle = tracing_init(
+    //         nullptr, 0,
+    //         initialFilter.c_str(),
+    //         false);
+    // }
```
Comment (daira) [inline src/test/test_bitcoin.cpp:88]: It would be nice to enable this by an environment variable (say `LOG_TO_STDOUT=1`).

**Review by daira — COMMENTED** [review]
Post-hoc ACK with comments (which I might file a follow-up PR for).

**Pull request merged**